server/server_test.go

```diff
@@ -40,10 +41,47 @@ func TestT(t *testing.T) {
 
 var regression = true
 
-var dsn = "root@tcp(localhost:4001)/test?strict=true"
+var defaultDSNConfig = map[string]interface{}{
+	"User":   "root",
+	"Net":    "tcp",
+	"Addr":   "127.0.0.1:4001",
+	"DBName": "test",
+	"Strict": true,
+}
+
+// getDSN generates a DSN string for MySQL connection based on
+// defaultDSNConfig and the given overrideConfig.
+func getDSN(overrideConfig map[string]interface{}) string {
```

Comment:
Personally, I dislike reflect.
How about this:

```
var defaultDSNConfig = mysql.Config {
    User : "root",
    Net  : "tcp",
    Addr : "127.0.0.1:4001",
    DBName : "test",
    Strict : true,
}

func getDSN(override func(mysql.Config) mysql.Config) string {
     return override(defaultDSNConfig).FormatDSN()
}
```

You see:

```
getDSN(func (X mysql.Config) mysql.Config {
            X.User = "xxx"
            X.Strict = false
            return X
})
getDSN(map[string]interface{}{
 	    "User":   "abc",
	    "Passwd": "wrong_password",
	    "DBName": "non_existing_schema",
	}
```

The former is not very complex and type safe, better performance, much like the Go way.